### PR TITLE
feat: add support for TypeScript 5.5

### PR DESCRIPTION
**Pull request by sjinks**
Update ESLint plugins to support TypeScript 5.5

### CODE REVIEW

File: __tests__/__snapshots__/check-fixtures.js.snap [hunk 1]
```diff
@@ -381,24 +381,6 @@ import { unusedEs6Import } from './stub/unusedEs6Import';
...
-      ],
-      "text": "// @ts-expect-error",
-    },
-    "line": 8,
-    "message": "Use "@ts-expect-error" to ensure an error is actually being suppressed.",
-    "messageId": "preferExpectErrorComment",
-    "nodeType": "Line",
-    "ruleId": "@typescript-eslint/prefer-ts-expect-error",
-    "severity": 2,
-  },
```
Comment (sjinks) [inline __tests__/__snapshots__/check-fixtures.js.snap:401]: https://github.com/typescript-eslint/typescript-eslint/pull/9081

The `@typescript-eslint/prefer-ts-expect-error` rule is deprecated; the bug is caught by `@typescript-eslint/ban-ts-comment` (line 360 of the snapshot), so we are good.

Ref: https://typescript-eslint.io/rules/prefer-ts-expect-error/

File: __tests__/__snapshots__/check-fixtures.js.snap [hunk 2]
```diff
@@ -458,12 +440,12 @@ import { unusedEs6Import } from './stub/unusedEs6Import';
   },
   {
     "column": 2,
-    "endColumn": 17,
+    "endColumn": 8,
```
Comment (sjinks) [inline __tests__/__snapshots__/check-fixtures.js.snap:443]: https://github.com/typescript-eslint/typescript-eslint/pull/8869

File: __tests__/__snapshots__/check-fixtures.js.snap [hunk 3]
```diff
@@ -513,12 +495,12 @@ import { unusedEs6Import } from './stub/unusedEs6Import';
   },
   {
     "column": 2,
-    "endColumn": 3,
-    "endLine": 48,
+    "endColumn": 24,
+    "endLine": 46,
```
Comment (sjinks) [inline __tests__/__snapshots__/check-fixtures.js.snap:499]: https://github.com/typescript-eslint/typescript-eslint/pull/8869

File: __tests__/__snapshots__/check-fixtures.js.snap [hunk 4]
```diff
@@ -513,12 +495,12 @@ import { unusedEs6Import } from './stub/unusedEs6Import';
...
     "line": 46,
     "message": "Missing accessibility modifier on method definition implicitlyPublicMethod.",
     "messageId": "missingAccessibility",
-    "nodeType": "MethodDefinition",
+    "nodeType": null,
```
Comment (sjinks) [inline __tests__/__snapshots__/check-fixtures.js.snap:503]: https://github.com/typescript-eslint/typescript-eslint/pull/8869/files#diff-1ff37ba3b572e256aa6c86ebc933b2207f82d588abdbe38a27376cca06bcb028L166

File: __tests__/__snapshots__/check-fixtures.js.snap [hunk 5]
```diff
@@ -458,12 +440,12 @@ import { unusedEs6Import } from './stub/unusedEs6Import';
...
   {
     "column": 2,
-    "endColumn": 17,
+    "endColumn": 8,
     "endLine": 38,
     "line": 38,
     "message": "Missing accessibility modifier on class property secret.",
     "messageId": "missingAccessibility",
-    "nodeType": "PropertyDefinition",
+    "nodeType": null,
```
Comment (sjinks) [inline __tests__/__snapshots__/check-fixtures.js.snap:448]: https://github.com/typescript-eslint/typescript-eslint/pull/8869/files#diff-1ff37ba3b572e256aa6c86ebc933b2207f82d588abdbe38a27376cca06bcb028L301

**Review by chriszarate — APPROVED** [review]

**Pull request merged**